### PR TITLE
Test runtime shaving

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

This PR cuts down on the test runtime significantly.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

# Additional comments

Fixes EDGE-383

### CODE REVIEW

**Review by Kourin1996 — APPROVED** [review]

File: archive/backup_test.go [hunk 1]
```diff
@@ -154,7 +156,10 @@ func Test_determineTo(t *testing.T) {
 	}
 
 	for _, tt := range tests {
```
Comment (0xAleksaOpacic) [inline archive/backup_test.go:158]: just curious, why do you need this?

File: jsonrpc/eth_state_test.go [hunk 2]
```diff
@@ -619,6 +619,8 @@ func getExampleStore() *mockSpecialStore {
 // the latest block gas limit for the upper bound, or the specified
 // gas limit in the transaction
 func TestEth_EstimateGas_GasLimit(t *testing.T) {
+	// TODO Make this test run in parallel when the race
```
Comment (0xAleksaOpacic) [inline jsonrpc/eth_state_test.go:622]: is this TODO leftover?

**Review by 0xAleksaOpacic — APPROVED** [review]

**Comment by 0xAleksaOpacic** [comment]
Can we remove this test and if needed add linear card for it
https://github.com/0xPolygon/polygon-edge/blob/293c8ef910034045c3870958cba16f3afdc15453/txpool/txpool_test.go#L1060

File: e2e/framework/helper.go [hunk 3]
```diff
@@ -29,6 +29,35 @@ import (
 	empty "google.golang.org/protobuf/types/known/emptypb"
 )
 
+var (
+	DefaultTimeout = time.Second * 10
```
Comment (0xAleksaOpacic) [inline e2e/framework/helper.go:33]: why 10 seconds to finish the test?

File: e2e/framework/helper.go [hunk 4]
```diff
@@ -429,18 +475,45 @@ func NewTestServers(t *testing.T, num int, conf func(*TestServerConfig)) []*Test
...
+		go func() {
+			defer wg.Done()
+
+			if err := srv.GenerateGenesis(); err != nil {
+				errors.Append(fmt.Errorf("server %d failed genesis command, error=%w", i, err))
+
+				return
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (0xAleksaOpacic) [inline e2e/framework/helper.go]: You can use DefaultTimeout for 10*time.Second

File: archive/backup_test.go (hunk 1 above)
Comment (0xAleksaOpacic) [reply archive/backup_test.go:158]: I found the reson https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721, good catch

**Comment by 0xAleksaOpacic** [comment]
When i have started tests locally i am getting the error. Tried it 2 times, can you please check this
--- FAIL: TestState (0.00s)
    state_test.go:122: open tests/GeneralStateTests: no such file or directory
--- FAIL: TestEVM (0.00s)
    evm_test.go:146: open tests/VMTests: no such file or directory
FAIL

**Comment by zivkovicmilos** [comment]
> When i have started tests locally i am getting the error. Tried it 2 times, can you please check this --- FAIL: TestState (0.00s) state_test.go:122: open tests/GeneralStateTests: no such file or directory --- FAIL: TestEVM (0.00s) evm_test.go:146: open tests/VMTests: no such file or directory FAIL

@Aleksao998 

Did you pull in the submodules?
`git submodule update --init --recursive` from the base of the repo

File: jsonrpc/eth_state_test.go (hunk 2 above)
Comment (zivkovicmilos) [reply jsonrpc/eth_state_test.go:622]: It's leftover for us to fix it when we fix `eth_estimateGas` 🙂 

There's a task up for it

**Comment by zivkovicmilos** [comment]
@Aleksao998 

Removed the unused `TestDemote`:

[3cd2919](https://github.com/0xPolygon/polygon-edge/pull/526/commits/3cd29198a1f78e51e40ed333013202c3bf75df45)

cc @dbrajovic

File: e2e/framework/helper.go (hunk 3 above)
Comment (zivkovicmilos) [reply e2e/framework/helper.go:33]: It's not used for tests, but for node joins:
https://github.com/0xPolygon/polygon-edge/blob/3cd29198a1f78e51e40ed333013202c3bf75df45/e2e/framework/helper.go#L248

File: e2e/framework/helper.go (hunk 4 above)
Comment (zivkovicmilos) [reply e2e/framework/helper.go]: I've swapped all occurrences where we use a 10s timeout for the `DefaultTimeout`:

[99d626e](https://github.com/0xPolygon/polygon-edge/pull/526/commits/99d626e69b20db13a7d5a7b1e28fb9b428aa25a3)

File: e2e/framework/helper.go (hunk 3 above)
Comment (zivkovicmilos) [reply e2e/framework/helper.go:33]: UPDATE:

I've swapped out all occurrences of the `time.second * 10` for the `DefaultTimeout`, check out a comment that's further down

**Review by ZeljkoBenovic — APPROVED** [review]
Great job guys 🏅 

I'm slapping this Approve so hard...

**Pull request merged**